src/utils/function.ts

```diff
@@ -0,0 +1,4 @@
+export const apply = <Input, Output>(
+  fnc: (v: Input) => Output,
+  value: Input | null | undefined,
+): Output | null | undefined => value === null || value === undefined ? null : fnc(value);
```

Comment:
Because it is a functor (https://dev.to/mpodlasin/functional-programming-in-js-functor-monad-s-little-brother-3053) so it preserves `undefined | null` and applies other values to the function.

Reply:
Oh, I get what you meant. It should return `undefined` when the input is `undefined`. You are right.